### PR TITLE
Bug 1159256 - [TaskSwitcher] Select icon size based on container size and device resolution

### PR DESCRIPTION
**Pull request by KevinGrandon**
https://bugzilla.mozilla.org/show_bug.cgi?id=1159256

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=12d3d47ae0e7f2f41cd87a03a7a0de4c52beff58

File: apps/system/js/cards_helper.js [hunk 1]
```diff
@@ -32,8 +23,16 @@
 
       sizes.sort(function(x, y) { return y - x; });
 
-      var index = sizes[(HVGA) ? sizes.length - 1 : 0];
-      iconPath = icons[index];
+      iconPath = icons[0]; // The biggest icon available
+      for (var i = 0; i < sizes.length; i++) {
+        var size = sizes[i];
+
+        if (size < maxSize) {
```
Comment (sfoster) [inline apps/system/js/cards_helper.js]: Should (couldnt) this be size <= maxSize? If size is 60 and there's a 60px icon available, I want to use it

Comment (KevinGrandon) [inline apps/system/js/cards_helper.js]: Well, you would still use it. If you have a <= then you would end up breaking, and not assigning the 60 image to the iconPath after the if statement. You would exit on the next iteration, and preserve the 60.

The logic is a bit confusing, if you have any ideas about how to clean it up, let me know.

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#Zzc7Sa8NRUCMjH1PytuvkA

The pull request failed to pass integration tests. It could not be landed, please try again.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=46cedad8d0a6d928e08cc05366ce77f3855dee44

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#aVPHYmNnTaen_x6WGWXCpQ

The pull request failed to pass integration tests. It could not be landed, please try again.

**Pull request merged**